### ISSUE TITLE
Consider security.tls.version.min 3 by default – for a minimum of TLS 1.2

### ISSUE THREAD

**Issue by grahamperrin** [MrAlex94/Waterfox#1288]
https://phabricator.services.mozilla.com/D45798#C1639160NL22 lines 22–26: 

```text
#ifdef RELEASE_OR_BETA
  pref("security.tls.version.min", 1);
#else
  pref("security.tls.version.min", 3);
#endif
```

Consider changing the default from `1` to `3` for both Waterfox Classic and Waterfox Current.

### Additional considerations

Mozilla bug [1579285 - Offer to re-enable TLS 1.0 and 1.1 on TLS version failure](https://bugzilla.mozilla.org/show_bug.cgi?id=1579285) 

- RESOLVED FIXED, Firefox 71
- I should not expect a comparable offer in Waterfox Classic or Waterfox Current
- advanced users may be directed to advice on manual setting of `security.tls.version.min`

From <https://github.com/MrAlex94/Waterfox/issues/783#issuecomment-499557922>: 

> … AFAIK, TLS 1.2 with a decent cipher is still considered secure, …

From <https://www.fxsitecompat.dev/en-CA/versions/71/>

> … Firefox 71 released on December 3, 2019. …

– and: 

> [TLS 1.0 and 1.1 are now deprecated, disabled in Nightly](https://www.fxsitecompat.dev/en-CA/docs/2019/tls-1-0-and-1-1-are-now-deprecated-disabled-in-nightly/)

– with reference to: 

* [Bug 1579270 - Disable TLS 1.0 and 1.1 for Nightly](https://bugzilla.mozilla.org/show_bug.cgi?id=1579270)
* [Intent to unship: TLS 1.0 and TLS 1.1](https://groups.google.com/d/topic/mozilla.dev.platform/8EFRYDR3N1c/discussion)
* [TLS 1.0 and 1.1 Removal Update - Mozilla Hacks](https://hacks.mozilla.org/2019/05/tls-1-0-and-1-1-removal-update/)

----

NB this issue is **not** about TLS 1.3 (there was a separate issue).

**Comment by grahamperrin** [comment]
To clarify: 

- not an immediate change
- consider changing the default **in due course** (ideally, in step with other browsers). 

Re: https://phabricator.services.mozilla.com/D45798#C1639160NL22 the *released* version of Firefox 71.0 does still still default to `1`. 

On one hand, there was the plan i.e.: 

> … Safari, Firefox, Edge and Chrome are removing support for TLS 1.0 and 1.1 in March of 2020. …

On the other hand: the reality of **many** sites not yet supporting TLS 1.2. 

----

I downloaded the latest Tranco list (22.3 MiB) from <https://tranco-list.eu/>, found 565 matches for `.ac.uk`. Focusing on two listings, the first of which was randomly selected: 

### `41242,nott.ac.uk`

<http://nott.ac.uk> redirects to <https://www.nottingham.ac.uk/> with a score of 10.0/10 in [SSleuth](https://github.com/sibiantony/ssleuth/).

<https://nott.ac.uk> is not suitably secure (and with reduced browser security, redirects to <https://nott.ac.uk/admin/index.php?u=https%3A%2F%2Fnott.ac.uk%2F1>).

### `16229,brighton.ac.uk` and `296721,bton.ac.uk`

Home, sweet home. 

For the staff intranet, much of which is intentionally made public (not entirely an intranet): 

![2019-12-04 10:40:16](https://user-images.githubusercontent.com/192271/70135931-e7e00800-1682-11ea-84f7-de2c397fa5eb.png)

* https://staff.brighton.ac.uk/is/Pages/itstatus.aspx using TLS 1.0

… and so on.

**Comment by hawkeye116477** [comment]
TLS 1.3 isn't supported by Waterfox Classic, it's from FF 63. It's also on Basilisk, so maybe could be ported..

**Comment by grahamperrin** [comment]
From Martin Thomson at <https://groups.google.com/d/msg/mozilla.dev.platform/8EFRYDR3N1c/njX0E7jhBAAJ>: 

>>> … Safari, Firefox, Edge and Chrome are removing support for TLS 1.0 and 1.1 in March of 2020. …
>>
>> Is that (timeline) still a *shared* intent – for all four browsers?
>
> I recently confirmed this, yes. 

– and: 

> The intent is to have that dialog to disappear some time after March next year or for the exceptions to be more narrowly targeted.  Re-enabling TLS 1.0 will still be possible via about:config for some time after that, so people with a real need can always turn it back on if they need to (including through enterprise policy if corporate services are stuck). That's still inadvisable; ultimately, we'd like to ensure that TLS 1.2 or higher is always used. 

@MrAlex94 would you like to close this issue for now, reopen in maybe two months' time?

**Comment by grahamperrin** [comment]
Recently released Firefox 74.0 has TLS 1.0 and 1.1 disabled by default, with an error message such as this for sites that fall below the preferred level: 

![image](https://user-images.githubusercontent.com/192271/77577276-6381f100-6ece-11ea-93c6-2e71d62c7db6.png)

**Comment by hawkeye116477** [comment]
@grahamperrin https://github.com/MrAlex94/Waterfox/commit/dc85e2eba41ba1818356422d361e81fd8ebbe444
